### ISSUE TITLE
Support for directories under pages

### ISSUE THREAD

**Issue by malimichael** [zeit/next.js#3508]
Is it currently possible to achieve a project structure like this? If not it would be great to add such a feature out of the box. 

```
-MyApp
    +components
    -pages
        -index
            index.js
            styles.js
            data.js
        -profile
            index.js
            styles.js
            data.js
```

**Comment by malimichael** [comment]
Ah it does indeed work 😄

**Comment by sergiodxa** [comment]
This was already discussed here #3183 and fixed #3195
